ONSdigital/eq-author-app
# Map our user type to a table.

### What is the context of this PR?
PR to add the concept of a user into our app.

This PR is dependent on https://github.com/ONSdigital/eq-author-terraform/pull/10 

Once this PR goes in the two new scripts `import-users-to-table.js` and `bulkMigrateDataStore.js` will both have to be run targeting all environments one by one.

### How to review the Import scripts

1. Checkout master.
2. Create a number of questionnaires with various users new and old.
3. Switch to this branch.
4. Change [this line](https://github.com/ONSdigital/eq-author-app/pull/334/files#diff-7719754e0e817167ed1622f31f045a22R7) to `let userTableName = "dev-author-users";` This is needed because the new env vars won't be picked up as the container will not be restarted.
4. Follow the instructions here: https://firebase.google.com/docs/cli/auth to export your existing users to a json file.
5. Add the default author team user to your firebase dump.
6. Run the `DYNAMO_QUESTIONNAIRE_TABLE_NAME=dev-author-questionnaires DYNAMO_QUESTIONNAIRE_VERSION_TABLE_NAME=dev-author-questionnaire-versions
DYNAMO_USER_TABLE_NAME=dev-author-users
AWS_ACCESS_KEY_ID=dummy AWS_SECRET_ACCESS_KEY=dummy DYNAMO_ENDPOINT_OVERRIDE=http://localhost:8050 AWS_REGION=eu-west-1 DATA_DIR=<path to firebase exports> node scripts/import-users-to-table.js` script with the firebase export.
7. Run the `DYNAMO_QUESTIONNAIRE_TABLE_NAME=dev-author-questionnaires DYNAMO_QUESTIONNAIRE_VERSION_TABLE_NAME=dev-author-questionnaire-versions
DYNAMO_USER_TABLE_NAME=dev-author-users AWS_ACCESS_KEY_ID=dummy AWS_SECRET_ACCESS_KEY=dummy DYNAMO_ENDPOINT_OVERRIDE=http://localhost:8050 AWS_REGION=eu-west-1 node bulkMigrateDataStore.js`script.
8. Restart the front-end to get the proxy in the webpack to work.
9. You should find all questionnaires have been updated to a user account, the user table has been correctly populated and any unclaimed users will be defaulted to 'Author Team'.

### How to review the rest.

After migrating to the new structure all questionnaires should be created and linked to a user account and in all other cases the app should work as expected.
Tests pass.
